massiv/src/Data/Massiv/Array/Delayed/Windowed.hs

```diff
@@ -23,16 +23,21 @@ import           Control.Monad                          (when)
 import           Data.Massiv.Array.Delayed.Internal
 import           Data.Massiv.Array.Manifest.BoxedStrict
 import           Data.Massiv.Array.Manifest.Internal
+import           Data.Massiv.Array.ValidateFunc
 import           Data.Massiv.Core
 import           Data.Massiv.Core.Common
 import           Data.Massiv.Core.List                  (showArray)
 import           Data.Massiv.Core.Scheduler
 import           Data.Proxy                             (Proxy (..))
 import           Data.Typeable                          (showsTypeRep, typeRep)
+import           Data.Validity
 
 -- | Delayed Windowed Array representation.
 data DW = DW
 
+instance Validity DW where
+    validate = trivialValidation
```

Comment:
This can be generated.

Reply:
Yes, but then I would have to add the "DeriveGenerics" language extension, import GHC.Generics, and derive a Generic description for DW. This sounded like a bad idea.
Do you agree?

Reply:
I don't agree, but it's not up to me.

Reply:
TLDR; @CSVdB, I'm ok with either approach. Same applies to all spots where @NorfairKing suggested generation.

I was thinking about adding Generic instance to most of the data types in massiv anyways, since it is pretty common for users of libraries to want them for various reasons. So I don't oppose to adding generic instances, but I don't find it important to switch to derived `Validity` instances for two reasons:
* you already have code that works and is semantically equivalent to the derived instances (at least I'd hope so ;)).
* Using derived mechanism does sometime marginally slow down compilation. Not terribly important.